geode-redis/src/distributedTest/java/org/apache/geode/redis/session/SessionDUnitTest.java

```diff
@@ -144,26 +146,51 @@ protected String createNewSessionWithNote(int sessionApp, String note) {
     HttpHeaders requestHeaders = new HttpHeaders();
     requestHeaders.add("Cookie", sessionCookie);
     HttpEntity<String> request2 = new HttpEntity<>("", requestHeaders);
-
-    return new RestTemplate()
-        .exchange(
-            "http://localhost:" + ports.get(sessionApp) + "/getSessionNotes",
-            HttpMethod.GET,
-            request2,
-            String[].class)
-        .getBody();
+    boolean sesssionObtained = false;
+    String[] responseBody = new String[0];
+    do {
+      try {
+        responseBody = new RestTemplate()
+            .exchange(
+                "http://localhost:" + ports.get(sessionApp) + "/getSessionNotes",
+                HttpMethod.GET,
+                request2,
+                String[].class)
+            .getBody();
+        sesssionObtained = true;
+      } catch (HttpServerErrorException e) {
+        if (e.getMessage().contains("Internal Server Error")) {
+          // retry
+        } else {
+          throw e;
+        }
+      }
+    } while (!sesssionObtained);
+    return responseBody;
```

Comment:
I'd prefer to see this retry logic happen in the Controller, but we can make that change later.

Reply:
Is the Controller part of the product or part of this test app?